btp-environment-variable-access artifacts bring slf4j-api 2.0.5

**Issue Summary**
- This is a follow up of #1036 
- We depend on `btp-environment-variable-access` which depends on `slf4j-api` 2.0.5, whereas we rollbacked this dependency to 1.7.36 in #1096
- As a result our library clients have a mix of `slf4j-api` 1.7.36 and 2.0.5 in their dependency tree when they bring our library, which can be a pain when using `maven-enforcer-plugin` to ensure correct transitive dependencies convergence
- We should remove the 2.0.5 version from our transitive dependencies

Sample:  clone or open in a codespace workspace this repository https://github.com/vtintillier/btp-environment-variable-access-111 and run `mvn validate`. It will fail with

```
[INFO] --- enforcer:3.1.0:enforce (default) @ demo ---
[ERROR] Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.slf4j:slf4j-api:1.7.36 paths to dependency are:
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security:java-api:2.13.6
      +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:1.7.36
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security.xsuaa:token-client:2.13.6
      +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:1.7.36
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security:env:2.13.6
      +-com.sap.cloud.environment.servicebinding:java-sap-vcap-services:0.5.2
        +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:2.0.5
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security:env:2.13.6
      +-com.sap.cloud.environment.servicebinding:java-sap-service-operator:0.5.2
        +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:2.0.5
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security:env:2.13.6
      +-com.sap.cloud.environment.servicebinding.api:java-access-api:0.5.2
        +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:2.0.5
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-com.sap.cloud.security:java-security:2.13.6
    +-com.sap.cloud.security.xsuaa:token-client:2.13.6
      +-ch.qos.logback:logback-classic:1.2.11 (managed) <-- ch.qos.logback:logback-classic:1.2.11
        +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:1.7.32
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-security:2.7.9
    +-org.springframework.boot:spring-boot-starter:2.7.9 (managed) <-- org.springframework.boot:spring-boot-starter:2.7.9
      +-org.springframework.boot:spring-boot-starter-logging:2.7.9 (managed) <-- org.springframework.boot:spring-boot-starter-logging:2.7.9
        +-org.apache.logging.log4j:log4j-to-slf4j:2.17.2 (managed) <-- org.apache.logging.log4j:log4j-to-slf4j:2.17.2
          +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:1.7.35
and
+-com.example:demo:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-security:2.7.9
    +-org.springframework.boot:spring-boot-starter:2.7.9 (managed) <-- org.springframework.boot:spring-boot-starter:2.7.9
      +-org.springframework.boot:spring-boot-starter-logging:2.7.9 (managed) <-- org.springframework.boot:spring-boot-starter-logging:2.7.9
        +-org.slf4j:jul-to-slf4j:1.7.36 (managed) <-- org.slf4j:jul-to-slf4j:1.7.36
          +-org.slf4j:slf4j-api:1.7.36 (managed) <-- org.slf4j:slf4j-api:1.7.36
```

<!-- Please make sure checklist is complete -->
**Issue Report Checklist**

- [x] Real, current bug
- [x] Not a duplicate  
- [x] Not covered in "Common Pitfalls" section of corresponding module's README.md e.g. [java-security#common-pitfalls](https://github.com/SAP/cloud-security-xsuaa-integration/tree/main/java-security#common-pitfalls)
- [x] Reproducible
- [x Good summary
- [x] Well-documented 
    - [ ] log level increased to `DEBUG` debug logs provided
    - [x] POM provided
    - [ ] dependency tree provided
    - [x] code snippet provided (when applicable)

--- a/env/pom.xml
+++ b/env/pom.xml
@@ -26,18 +26,46 @@
         <dependency>
             <groupId>com.sap.cloud.environment.servicebinding</groupId>
             <artifactId>java-sap-vcap-services</artifactId>
+            <!-- it depends on slf4j-api 2.x whereas we depends on 1.7.x -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>com.sap.cloud.environment.servicebinding</groupId>
             <artifactId>java-sap-service-operator</artifactId>
+            <!-- it depends on slf4j-api 2.x whereas we depends on 1.7.x -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency><!-- optional if using TypedMapView etc. -->
             <groupId>com.sap.cloud.environment.servicebinding.api</groupId>
             <artifactId>java-consumption-api</artifactId>
+            <!-- it depends on slf4j-api 2.x whereas we depends on 1.7.x -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>com.sap.cloud.environment.servicebinding.api</groupId>
             <artifactId>java-access-api</artifactId>
+            <!-- it depends on slf4j-api 2.x whereas we depends on 1.7.x -->
+            <exclusions>
+                <exclusion>
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>slf4j-api</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>org.json</groupId>

